programs/exchange/src/lib.rs

```diff
@@ -1778,15 +1800,18 @@ pub mod exchange {
             val: amount.into(),
             scale: synthetic.supply.scale,
         };
-        // fee is removed from synthetic supply
-        // this causes pro rata distribution to stakers
         let fee = amount.mul(swapline.fee);
         let amount_out = amount.sub(fee).unwrap().to_scale(swapline.balance.scale);
         let new_supply_synthetic = synthetic.supply.sub(amount).unwrap();
```

Comment:
missing check in case there are not enough collaterals for swap. Probably this line `synthetic.swapline_supply = synthetic.swapline_supply.sub(amount).unwrap();` will revert transaction in that case. Maybe it will be good idea to add require and error code.

Reply:
This should be handled by the UI, but we can also add explicit error.